src/RequestHandler/outofprocess/forwardinghandler.cpp

```diff
@@ -1252,27 +1321,41 @@ None
     if (TlsGetValue(g_dwTlsIndex) != this)
     {
         DBG_ASSERT(TlsGetValue(g_dwTlsIndex) == NULL);
+        if (m_RequestStatus != FORWARDER_RECEIVED_WEBSOCKET_RESPONSE)
+        {
+            // Webscoket has already been guarded by critical section
+            // Only require exclisive lock for non-websocket scenario which has duplex channel
+            // Otherwise, there will be a deadlock
+            AcquireSRWLockExclusive(&m_RequestLock);
```

Comment:
Make helper functions to lock and unlock as we do it 5 times throughout the file? Like:
```c++
VOID
FORWARDING_HANDLER::LockExclusive()
 {
            AcquireSRWLockExclusive(&m_RequestLock);
            TlsSetValue(g_dwTlsIndex, this);
            fExclusiveLocked = TRUE;
            DBG_ASSERT(TlsGetValue(g_dwTlsIndex) == this);
}
```

Same for releasing and shared locks.